### PR TITLE
:bug: Source Jira: `check_connection` fails if no projects are defined

### PR DESCRIPTION
**Pull request by philippeboyd**
## What
During connection check, the `check_connection(...)` method would fail because the key `projects` in `config` is not set **if the user doesn't specify any project**; we get the following error:

```
The connection tests failed.
KeyError('projects')
```

image

A simple workaround is to specify a dummy project and delete it so that the array can be initialized (`[]`).

## How
Default to empty array if the key `projects` is not in `config`.
This fix is simply to avoid the unnecessary workaround.

### CODE REVIEW

**Comment by philippeboyd** [comment]
Hi @grubberr , let me know if this makes sense to you. Thanks!

File: airbyte-integrations/connectors/source-jira/source_jira/source.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def get_authenticator(config: Mapping[str, Any]):
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
         config = self._validate_and_transform(config)
         authenticator = self.get_authenticator(config)
-        kwargs = {"authenticator": authenticator, "domain": config["domain"], "projects": config["projects"]}
+        kwargs = {"authenticator": authenticator, "domain": config["domain"], "projects": config.get("projects", [])}
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-jira/source_jira/source.py:84]: can you please move this to `_validate_and_transform` function and make such transformation:

```
config["projects"] = config.get("projects", [])
```

?

**Review by grubberr — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-jira/source_jira/source.py (hunk 1 above)
Comment (philippeboyd) [reply airbyte-integrations/connectors/source-jira/source_jira/source.py]: Done :+1:

**Review by grubberr — APPROVED** [review]

**Comment by philippeboyd** [comment]
@grubberr makes sense, see you next year then. Thanks for the quick review and happy holidays!

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 2]
```diff
@@ -785,7 +785,7 @@
 - name: Jira
   sourceDefinitionId: 68e63de2-bb83-4c7e-93fa-a8a9051e3993
   dockerRepository: airbyte/source-jira
-  dockerImageTag: 0.3.1
+  dockerImageTag: 0.3.2
```
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:788]: revert this change, it will be bumped automatically by publishing scripts

**Review by grubberr — CHANGES_REQUESTED** [review]

File: airbyte-config/init/src/main/resources/seed/source_specs.yaml [hunk 3]
```diff
@@ -6668,7 +6668,7 @@
     supportsNormalization: false
     supportsDBT: false
     supported_destination_sync_modes: []
-- dockerImage: "airbyte/source-jira:0.3.1"
+- dockerImage: "airbyte/source-jira:0.3.2"
```
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_specs.yaml:6671]: revert this change, it will be bumped automatically by publishing scripts

**Review by grubberr — CHANGES_REQUESTED** [review]

**Comment by grubberr** [comment]
@philippeboyd please merge master branch to your branch because upstream code changed

File: airbyte-config/init/src/main/resources/seed/source_specs.yaml (hunk 3 above)
Comment (philippeboyd) [reply airbyte-config/init/src/main/resources/seed/source_specs.yaml]: Done :heavy_check_mark:

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml (hunk 2 above)
Comment (philippeboyd) [reply airbyte-config/init/src/main/resources/seed/source_definitions.yaml]: Done :heavy_check_mark:

**Comment by philippeboyd** [comment]
> @philippeboyd please merge master branch to your branch because upstream code changed

@grubberr Done :heavy_check_mark:

**Review by grubberr — APPROVED** [review]

**Comment by grubberr** [comment]
@alafanechere @evantahler Can we merge this to master? 
I have already published this, it's pretty simple PR

**Comment by alafanechere** [comment]
@grubberr I'd prefer not to merge until we have 🟢 checks. 
I pinged pipeline folks to understand why the required checks are not running.
Let me know if it's blocking other Jira improvements, I'll delete the new docker image on docker hub.

**Comment by grubberr** [comment]
@alafanechere ok not big deal we can wait

**Comment by philippeboyd** [comment]
@alafanechere any news from the pipeline folks?

**Comment by evantahler** [comment]
trying again...
